src/azure-cli/azure/cli/command_modules/netappfiles/custom.py

```diff
@@ -386,18 +404,31 @@ def remove_export_policy_rule(instance, rule_index):
 
 
 # ---- SNAPSHOTS ----
-def create_snapshot(client, resource_group_name, account_name, pool_name, volume_name, snapshot_name, location):
+def create_snapshot(client, resource_group_name, account_name, pool_name, volume_name, snapshot_name, location=None):
+    if location is None:
+        location = client.resource_groups.get(resource_group_name).location
     body = Snapshot(location=location)
     return client.begin_create(resource_group_name, account_name, pool_name, volume_name, snapshot_name, body)
 
 
+def snapshot_restore_files(client, resource_group_name, account_name, pool_name, volume_name, snapshot_name, file_paths,
+                           destination_path=None):
+    body = SnapshotRestoreFiles(
+        file_paths=file_paths,
+        destination_path=destination_path
+    )
+    client.begin_restore_files(resource_group_name, account_name, pool_name, volume_name, snapshot_name, body)
```

Comment:
The client method start with `begin_` prefix is long running operation. Why not add --no-wait for those kind of command?